app/models/reports/repository.rb

```diff
@@ -16,112 +23,55 @@ def initialize(regions, periods:, reporting_schema_v2: false)
       @period_type = @periods.first.type
       @reporting_schema_v2 = reporting_schema_v2
       raise ArgumentError, "Quarter periods not supported" if @period_type != :month
+      @schema = if reporting_schema_v2?
```

Comment:
This is the "schema" or maybe "data access object" that either grabs things from the v1 matviews or the new v2 ones.